dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/event/DateValidator.java

```diff
@@ -70,25 +70,28 @@ public void validate(Reporter reporter, TrackerBundle bundle, Event event) {
       return;
     }
 
+    validateCompletedDateIsSetOnlyForSupportedStatus(reporter, event);
     validateExpiryDays(reporter, event, program, bundle.getUser());
     validatePeriodType(reporter, event, program);
   }
 
+  private void validateCompletedDateIsSetOnlyForSupportedStatus(Reporter reporter, Event event) {
+    if (event.getCompletedAt() != null && EventStatus.COMPLETED != event.getStatus()) {
+      reporter.addError(event, E1051, event, event.getStatus());
+    }
+  }
+
   private void validateExpiryDays(
       Reporter reporter, Event event, Program program, UserDetails user) {
-    if (user.isAuthorized(Authorities.F_EDIT_EXPIRED.name())) {
+    if (user.isAuthorized(Authorities.F_EDIT_EXPIRED.name()) || event.getCompletedAt() == null) {
```

Comment:
```suggestion
    if (event.getCompletedAt() == null || user.isAuthorized(Authorities.F_EDIT_EXPIRED.name())) {
```

I assume `isAuthorized` needs to check some collection which can be avoided if the date is null.